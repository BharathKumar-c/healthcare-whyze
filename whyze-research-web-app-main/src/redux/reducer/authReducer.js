import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
