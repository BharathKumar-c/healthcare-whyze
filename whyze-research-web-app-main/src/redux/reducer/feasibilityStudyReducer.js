import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
