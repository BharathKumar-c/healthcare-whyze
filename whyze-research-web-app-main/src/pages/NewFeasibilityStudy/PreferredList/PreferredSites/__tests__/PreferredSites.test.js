import { fireEvent, render } from '@testing-library/react';
import PreferredSitesTab from '..';
import React from 'react';
import Store from '../../../../../redux/store';
import { Provider } from 'react-redux';
import { act } from 'react-dom/test-utils';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={Store}>{element}</Provider>,
  );
  return { container, Store, getByText };
}
const demoSites = {
  id: 3,
  isChecked: true,
  isLinked: false,
  location: 'London, UK',
  noOfConnectedSites: 87,
  noOfNotConnectedSites: 6,
  star: 4.1,
  university: 'University College Hospital at Westmoreland Street',
};
const demoWithLinkedSites = {
  id: 4,
  isChecked: false,
  isLinked: true,
  location: 'Cork, Ireland',
  noOfLinks: 1,
  linkedData: [
    {
      location: 'Dublin, Ireland',
      name: 'Cork University',
      noOfConnectedSites: 35,
      noOfNotConnectedSites: 11,
      star: 3.9,
    },
  ],
  noOfConnectedSites: 87,
  noOfNotConnectedSites: 7,
  star: 3.7,
  university: 'Cork University Hospital',
};
const demoWithmanyLinkendSites = {
  id: 4,
  isChecked: false,
  isLinked: true,
  location: 'Cork, Ireland',
  noOfLinks: 3,
  linkedData: [
    {
      location: 'Dublin, Ireland',
      name: 'Cork University',
      noOfConnectedSites: 35,
      noOfNotConnectedSites: 11,
      star: 3.9,
    },
    {
      location: 'Dublin, Ireland',
      name: 'Cork University',
      noOfConnectedSites: 35,
      noOfNotConnectedSites: 11,
      star: 3.9,
    },
  ],
  noOfConnectedSites: 87,
  noOfNotConnectedSites: 7,
  star: 3.7,
  university: 'Cork University Hospital',
};
describe('test cases for PreferredSitesTab ', () => {
  test('should container render check', () => {
    const { container } = renderWithContext(<PreferredSitesTab />);
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].preferredSites.push(demoSites);

      Store.dispatch(setSegments(segments));
    });
    expect(container).toMatchSnapshot();
  });
  test('should render  onChange tab corrctly', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    expect(container).toBeInTheDocument();
  });
  test('should checkBox trigger event when click', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const Checkbox = container.getElementsByClassName('handleChecked-checkbox');

    fireEvent.click(Checkbox[0].firstChild);
  });
  test('should linked sites expand click & render all linked sites count ', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].preferredSites.push(demoWithLinkedSites);

      Store.dispatch(setSegments(segments));
    });
    const Checkbox = container.getElementsByClassName(
      'preferred-list-children-wrapper-linked',
    );

    fireEvent.click(Checkbox[0].firstChild);
    expect(Checkbox[0].lastChild.textContent).toBe('1 linked site');
  });
  test('should linked sites expand click & render all linked sites many count ', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].preferredSites.push(
        demoWithmanyLinkendSites,
      );

      Store.dispatch(setSegments(segments));
    });
    const Checkbox = container.getElementsByClassName(
      'preferred-list-children-wrapper-linked',
    );

    fireEvent.click(Checkbox[0].firstChild);
    expect(Checkbox[1].lastChild.textContent).toBe('3 linked sites');
  });
  test('should delete data the data in preferredSites', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'handleChecked-checkbox',
    );

    fireEvent.click(preferredListCheckbox[0]);
  });
  test('should checked single data', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );

    fireEvent.click(preferredListCheckbox[0]);
  });
  test('should sitesContacted-trigger clicked', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'preferred-list-checkbox',
    );

    fireEvent.click(preferredListCheckbox[0].firstChild);
    const preferredList = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );

    fireEvent.click(preferredList[2].firstChild);
  });
  test('should sitesInitiated-trigger clicked', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'preferred-list-checkbox',
    );

    fireEvent.click(preferredListCheckbox[0].firstChild);
    const preferredList = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );

    fireEvent.click(preferredList[1].firstChild);
  });
  test('should expand the sites using Arrow key down', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const DownArrow = container.getElementsByClassName('ArrowDown-toggle');

    fireEvent.click(DownArrow[0].firstChild);
    expect(DownArrow[0].firstChild.textContent).toBe('Dropdown.svg');
    const UpArrow = container.getElementsByClassName('ArrowUp-toggle');

    fireEvent.click(UpArrow[0].firstChild);
  });

  test('should checked all data in preferredSites', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'preferred-list-checkbox',
    );

    fireEvent.click(preferredListCheckbox[0].firstChild);
    expect(container).toBeInTheDocument();
  });
  test('should checked onClickCheckedContactedData', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'sitesContacted-trigger',
    );

    fireEvent.click(preferredListCheckbox[0].firstChild);
  });
  test('should checked onClickCheckedInitiatedData', () => {
    const { container } = renderWithContext(
      <PreferredSitesTab
        onTabChange={() => {
          'sitesContacted';
        }}
      />,
    );
    const preferredListCheckbox = container.getElementsByClassName(
      'sitesInitiated-trigger',
    );

    fireEvent.click(preferredListCheckbox[0].firstChild);
  });
});
