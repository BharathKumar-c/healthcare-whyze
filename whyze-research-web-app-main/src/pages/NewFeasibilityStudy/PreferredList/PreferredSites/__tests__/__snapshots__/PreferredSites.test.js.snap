// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for PreferredSitesTab  should container render check 1`] = `
<div>
  <div
    class="preferred-list-pagination-container"
  >
    <div
      class="preferred-list-alignment"
    >
      <div
        class="preferred-list-alignment"
      >
        <div
          class="preferred-list-checkbox"
        >
          <label
            class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                class="ant-checkbox-input"
                name="sites"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span
            class="ant-typography"
            style="margin-left: 1rem;"
          >
            1 Sites
          </span>
        </div>
        <div
          class="ant-divider ant-divider-vertical preferred-list-divider"
          role="separator"
        />
      </div>
    </div>
    <div
      class="preferred-list-options"
    >
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Delete.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Delete
          </span>
        </div>
      </div>
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Contact.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Mark contacted
          </span>
        </div>
      </div>
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Initiated.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Mark initiated
          </span>
        </div>
      </div>
    </div>
    <div
      class="preferred-list-alignment"
    >
      <span
        class="ant-typography preferred-list-pagination-text"
      >
        1-1 of 1
      </span>
      <div
        class="preferred-list-arrow-icon"
      >
        <svg
          class="preferred-list-arrow-left"
        >
          LeftArrow.svg
        </svg>
        <svg
          class="preferred-list-arrow-right"
        >
          RightArrow.svg
        </svg>
      </div>
    </div>
  </div>
  <div
    class="preferred-list-children"
  >
    <div
      class="ant-col ant-col-md-1"
    >
      <div
        class="handleChecked-checkbox"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-typography"
          style="margin-left: 1rem;"
        />
      </div>
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center selected-condition-container"
      style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
    >
      <div
        class="ant-space-item"
      >
        <div
          class="ant-row preferred-list-children-wrapper"
        >
          <div
            class="ant-col"
          >
            <div
              class="ant-row preferred-list-children-wrapper-university"
            >
              University College Hospital at Westmoreland Street
            </div>
            <div
              class="ant-row preferred-list-children-wrapper-usersAndLocation"
            >
              <div
                class="ant-col preferred-list-children-wrapper-usersAndLocation-col"
              >
                <svg>
                  UserCheck.svg
                </svg>
                87
              </div>
              <svg
                class="preferred-list-children-wrapper-usersAndLocation-col-icon"
              >
                Divider.svg
              </svg>
              <div
                class="ant-col preferred-list-children-wrapper-usersAndLocation-col"
              >
                <svg>
                  UserDelete.svg
                </svg>
                6
              </div>
              <svg
                class="preferred-list-children-wrapper-usersAndLocation-col-icon"
              >
                Divider.svg
              </svg>
              <div
                class="ant-col preferred-list-children-wrapper-usersAndLocation-col"
              >
                4.1
                <svg>
                  Star.svg
                </svg>
              </div>
              <svg
                class="preferred-list-children-wrapper-usersAndLocation-col-icon"
              >
                Divider.svg
              </svg>
              <div
                class="ant-col preferred-list-children-wrapper-usersAndLocation-col"
              >
                <svg>
                  Location2.svg
                </svg>
                London, UK
              </div>
            </div>
          </div>
          <div
            class="ant-col preferred-list-children-wrapper-info"
          >
            <div
              class="sitesContacted-trigger"
            >
              <svg>
                Contact.svg
              </svg>
            </div>
            <div
              class="sitesInitiated-trigger"
            >
              <svg>
                Initiated.svg
              </svg>
            </div>
            <div
              class="toggles-for-expand"
            >
              <svg
                class="ArrowDown-toggle"
              >
                Dropdown.svg
              </svg>
            </div>
          </div>
          <div
            class="content"
          >
            <div
              class="modal-children-container"
            >
              <div
                class="modal-children-container_patient-segment-container"
              >
                <h4>
                  Patient segment
                </h4>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-12"
                  >
                    <div
                      class="Connectsites"
                    >
                      <svg>
                        connectsites.svg
                      </svg>
                      <h3>
                        87
                      </h3>
                    </div>
                    <p>
                      Connected with this site
                    </p>
                  </div>
                  <div
                    class="ant-col ant-col-12"
                  >
                    <div
                      class="Catchmentsites"
                    >
                      <svg>
                        catchmentsites.svg
                      </svg>
                      <h3>
                        6
                      </h3>
                    </div>
                    <p>
                      Additional Whyze patients in site catchment areas
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="modal-children-container_patient-segment-rating"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-8"
                  >
                    <div
                      class="Rating"
                    >
                      <h3>
                        4.1
                      </h3>
                      <ul
                        class="ant-rate"
                        role="radiogroup"
                        tabindex="0"
                      >
                        <li
                          class="ant-rate-star ant-rate-star-full"
                        >
                          <div
                            aria-checked="true"
                            aria-posinset="1"
                            aria-setsize="5"
                            role="radio"
                            tabindex="0"
                          >
                            <div
                              class="ant-rate-star-first"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="ant-rate-star-second"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </li>
                        <li
                          class="ant-rate-star ant-rate-star-full"
                        >
                          <div
                            aria-checked="true"
                            aria-posinset="2"
                            aria-setsize="5"
                            role="radio"
                            tabindex="0"
                          >
                            <div
                              class="ant-rate-star-first"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="ant-rate-star-second"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </li>
                        <li
                          class="ant-rate-star ant-rate-star-full"
                        >
                          <div
                            aria-checked="true"
                            aria-posinset="3"
                            aria-setsize="5"
                            role="radio"
                            tabindex="0"
                          >
                            <div
                              class="ant-rate-star-first"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="ant-rate-star-second"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </li>
                        <li
                          class="ant-rate-star ant-rate-star-full"
                        >
                          <div
                            aria-checked="true"
                            aria-posinset="4"
                            aria-setsize="5"
                            role="radio"
                            tabindex="0"
                          >
                            <div
                              class="ant-rate-star-first"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="ant-rate-star-second"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </li>
                        <li
                          class="ant-rate-star ant-rate-star-zero"
                        >
                          <div
                            aria-checked="true"
                            aria-posinset="5"
                            aria-setsize="5"
                            role="radio"
                            tabindex="0"
                          >
                            <div
                              class="ant-rate-star-first"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="ant-rate-star-second"
                            >
                              <span
                                aria-label="star"
                                class="anticon anticon-star"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="star"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-8"
                  >
                    <div
                      class="Rating"
                    >
                      <h3>
                        0.0 months
                      </h3>
                      <p>
                        Recruitment timeline
                      </p>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-8"
                  >
                    <div
                      class="Rating"
                    >
                      <h3>
                         
                        <svg>
                          Trending.svg
                        </svg>
                        00.00%
                      </h3>
                      <p>
                        Drop out rate
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-children-container_track-recoder-container"
            >
              <h4>
                Track record of doing trials
              </h4>
              <div
                class="ant-row"
              >
                <div
                  class="ant-col rating-wrapper"
                >
                  <div>
                    <h3>
                       
                      <svg>
                        Trending.svg
                      </svg>
                      XX trials
                    </h3>
                  </div>
                  <p>
                    Completed in the past year
                  </p>
                </div>
              </div>
            </div>
            <div
              class="modal-children-container_details-container"
            >
              <div
                class="ant-row"
              >
                <div
                  class="ant-col ant-col-12"
                >
                  <div
                    class="Primary-Investigator"
                  >
                    <h3>
                      Name Surname
                    </h3>
                  </div>
                  <p>
                    Primary Investigator
                  </p>
                  <h4>
                    <svg>
                      Phone.svg
                    </svg>
                    Phone number
                  </h4>
                  <h4>
                    <svg>
                      world-2.svg
                    </svg>
                    Email address
                  </h4>
                </div>
                <div
                  class="ant-col ant-col-12"
                >
                  <div
                    class="Administration-Investigator"
                  >
                    <h3>
                      Name Surname
                    </h3>
                  </div>
                  <p>
                    Administration
                  </p>
                  <h4>
                    <svg>
                      Phone.svg
                    </svg>
                    Phone number
                  </h4>
                  <h4>
                    <svg>
                      world-2.svg
                    </svg>
                    Email address
                  </h4>
                </div>
              </div>
            </div>
            <div
              class="modal-children-container_industry-avarage-rating"
            >
              <div
                class="ant-row"
              >
                <div
                  class="ant-col ant-col-6"
                >
                  <div>
                    <h3>
                       
                      <svg>
                        Trending.svg
                      </svg>
                      00.00%
                    </h3>
                    <p>
                      Screen fail rate
                    </p>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-6"
                >
                  <div>
                    <span
                      class="ant-tag ant-tag-error tag"
                    >
                      <svg>
                        Trending.svg
                      </svg>
                      <span>
                         
                        26.7%
                      </span>
                    </span>
                    <p>
                      Industry average
                    </p>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-6"
                >
                  <div>
                    <h3>
                       
                      <svg>
                        Trending.svg
                      </svg>
                      00.00%
                    </h3>
                    <p>
                      Avg. response time
                    </p>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-6"
                >
                  <div>
                    <span
                      class="ant-tag ant-tag-error tag"
                    >
                      <svg>
                        Trending.svg
                      </svg>
                      <span>
                         
                        2.5 months
                      </span>
                    </span>
                    <p>
                      Industry average
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
